### PR TITLE
Add possibility of overwriting app id

### PR DESCRIPTION
**Pull request by pchojnacki**
- add tests to app creation from stage definition

### CODE REVIEW

File: marathon-gradle-plugin/build.gradle [hunk 1]
```diff
@@ -3,8 +3,7 @@ apply from: rootProject.file('gradle/deploy.gradle')
 
 group = 'com.wikia.gradle'
 sourceCompatibility = 1.7
-version = "0.5.36"
-
+version = "0.5.37"
```
Comment (pchojnacki) [inline marathon-gradle-plugin/build.gradle]: test

File: marathon-gradle-plugin/build.gradle [hunk 2]
```diff
@@ -3,8 +3,7 @@ apply from: rootProject.file('gradle/deploy.gradle')
...
 
 repositories {
     maven { url "http://repo.jenkins-ci.org/repo" }
```
Comment (pchojnacki) [inline marathon-gradle-plugin/build.gradle]: test2

File: marathon-gradle-plugin/src/main/groovy/com/wikia/gradle/marathon/AppFactory.groovy [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        app.setLabels(marathon.resolveLabels().labels)
+
+        app.setBackoffFactor(marathon.backoffFactor)
+        app.setBackoffSeconds(marathon.backoffSeconds)
+        app.setMaxLaunchDelaySeconds(marathon.maxLaunchDelaySeconds)
+
+        List<HealthCheck> healthChecks = stage.resolve(Healthchecks).healthchecksProvider()
+        app.setHealthChecks(healthChecks)
+
+        List<List<String>> constraints = stage.resolve(Constraints).getConstraints()
```
Comment (sqreek) [inline marathon-gradle-plugin/src/main/groovy/com/wikia/gradle/marathon/AppFactory.groovy]: List of list is necessary?

**Comment by pchojnacki** [comment]
LGTMized via upsource

File: marathon-gradle-plugin/src/main/groovy/com/wikia/gradle/marathon/AppFactory.groovy (hunk 3 above)
Comment (pchojnacki) [inline marathon-gradle-plugin/src/main/groovy/com/wikia/gradle/marathon/AppFactory.groovy]: its required by upstream api which is a list of lists

**Pull request merged**